File: an2_sem4/MPP/Homework/lab2/cs/FestivalSellpoint/FestivalSellpoint.Repo/AngajatDbRepo.cs
using FestivalSellpoint.Domain;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FestivalSellpoint.Repo
{
    public class AngajatDbRepo : DatabaseRepoUtils<int, Angajat>, IAngajatRepo
    {
        public AngajatDbRepo(IDictionary<string, string> props) : base(props)
        {
        }

        public void Add(Angajat e)
        {
            int result = ExecuteNonQuery("insert into \"Angajat\" (\"username\", \"password\", \"email\") " +
                "values (@username, @password, @email)", new Dictionary<string, object>
                {
                    { "@username", e.Username },
                    { "@password", e.Password },
                    { "@email", e.Email }
                });
            if (result == 0)
                throw new RepositoryException("Angajat was not added");
        }

        public Angajat findByCredentials(string username, string password)
        {
            return SelectFirst("select * from \"Angajat\" where " +
                "\"username\"=@username and \"password\"=@password",
                new Dictionary<string, object>
                {
                    { "@username", username },
                    { "@password", password },                    
                });
        }

        public IEnumerable<Angajat> GetAll()
        {
            return Select("select * from \"Angajat\"");
        }

        public Angajat GetById(int id)
        {
            return SelectFirst("select * from \"Angajat\" where \"id\"=@id", new Dictionary<string, object>
            {
                { "@id", id },
            });
        }

        public void Remove(int id)
        {
            ExecuteNonQuery("delete from \"Angajat\" where \"id\"=@id", new Dictionary<string, object>
            {
                { "@id", id },
            });
        }

        public void Update(Angajat e)
        {
            throw new RepositoryException("Angajat update is not allowed");
        }

        protected override Angajat DecodeReader(IDataReader reader)
        {
            var x = reader["id"];
            var id = Convert.ToInt32(reader["id"]);
            var username = reader["username"] as string;
            var password = reader["password"] as string;
            var email = reader["email"] as string;            
            var angajat = new Angajat(username, password, email);
            angajat.Id = id;
            return angajat;
        }
    }
}

File: an2_sem4/MPP/Homework/sapt78/java/repo/src/main/java/ro/ubbcluj/cs/stefnmubb/festivalsellpoint/repo/AngajatDbRepo.java
package ro.ubbcluj.cs.stefnmubb.festivalsellpoint.repo;

import org.apache.logging.log4j.Logger;
import org.apache.logging.log4j.LogManager;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Angajat;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Properties;
import java.util.stream.StreamSupport;

public class AngajatDbRepo extends DatabaseRepoUtils<Integer, Angajat> implements IAngajatRepo {
    private static final Logger logger= LogManager.getLogger(AngajatDbRepo.class);

    public AngajatDbRepo(Properties props){
        super(props);
        logger.info("Initializing AngajatDbRepo with properties: {} ",props);
    }

    @Override
    public Angajat decodeResult(ResultSet result) throws SQLException {
        var id = result.getInt("id");
        var username = result.getString("username");
        var password = result.getString("password");
        var email = result.getString("email");
        var angajat = new Angajat(username, password, email);
        angajat.setId(id);
        return angajat;
    }

    @Override
    public Angajat findByCredentials(String username, String password) throws EntityRepoException {
        return selectFirst("select * from \"Angajat\" where " +
                "\"username\"=? and \"password\"=?"
                ,username, password);
    }

    @Override
    public void add(Angajat angajat) throws EntityRepoException {
        logger.trace("Inserting {}", angajat);

        executeNonQuery("insert into \"Angajat\" (\"username\", \"password\", \"email\") values (?, ?, ?)",
                angajat.getUsername(),
                angajat.getPassword(),
                angajat.getEmail());

        logger.info("Inserted successfully");
        logger.traceExit();
    }

    @Override
    public void update(Angajat angajat) throws EntityRepoException {
        throw new EntityRepoException("Angajat update is not allowed");
    }

    @Override
    public void remove(Integer integer) throws EntityRepoException {
        throw new EntityRepoException("Angajat removal is not allowed");
    }

    @Override
    public Angajat getById(Integer id) throws EntityRepoException {
        return selectFirst("select * from \"Angajat\" where \"id\"=?",id);
    }

    @Override
    public Iterable<Angajat> getAll() throws EntityRepoException {
        return select("select * from \"Angajat\"");
    }
}
